### PR TITLE
BaSP-TG-74: Project List 

### PR DESCRIPTION
**Pull request by laura-brussa12**
image

**Project List:**

- Implement the shared componet "Table"
-  Implement the shared componet "Modal"
-  Implement the shared componet "Button"
-  Implement the shared componet "Loading Screen"
- And Serch-bar

### CODE REVIEW

File: src/Components/Projects/index.js [hunk 1]
```diff
@@ -1,70 +1,90 @@
...
-      try {
-        await fetch(`${process.env.REACT_APP_API_URL}/projects/${projectId}`, {
-          method: 'DELETE'
-        });
-        const result = projects.filter((project) => project._id !== projectId);
-        setProjects(result);
-      } catch (error) {
-        console.log(error);
-      }
+  const deleteItem = async (_id) => {
```
Comment (AlejandroSantini) [inline src/Components/Projects/index.js:37]: The function does not need to be asynchronous

File: src/Components/Projects/index.js [hunk 2]
```diff
@@ -1,70 +1,90 @@
...
+      setIdToDelete(_id);
+      setLoading(false);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  async function deleteProject() {
+    try {
+      await fetch(`${process.env.REACT_APP_API_URL}/projects/${idToDelete}`, {
```
Comment (AlejandroSantini) [inline src/Components/Projects/index.js:50]: You should add setLoading (true), then setLoading (false)

File: src/Components/Shared/LoadingScreen/ls.module.css [hunk 3]
```diff
@@ -1,9 +1,6 @@
 .background {
   bottom: 0;
   left: 0;
-  display: flex;
-  flex-direction: column;
```
Comment (AlejandroSantini) [inline src/Components/Shared/LoadingScreen/ls.module.css:5]: What are these changes for?

**Review by AlejandroSantini — APPROVED** [review]
Nice job!

**Review by AlejandroSantini — DISMISSED** [review]
Correct comments

File: src/Components/Projects/index.js [hunk 4]
```diff
@@ -1,70 +1,90 @@
...
   return (
     <section className={styles.container}>
-      <h2>- Projects List -</h2>
+      <Search
+        searchQuery={search}
+        setSearchQuery={setSearchProject}
+        placeholder="Search for Projects"
+      />
+      <div className={styles.add}>
+        <Button link={'/projects/form'} text={'Add Form'}></Button>
```
Comment (CarVigil) [inline src/Components/Projects/index.js:72]: Button must be on the left and the text should be "Add Project"

File: src/Components/Projects/index.js [hunk 5]
```diff
@@ -1,70 +1,90 @@
...
   }
 
+  const setSearchProject = (value) => {
+    setSearch(value);
+  };
+
   return (
     <section className={styles.container}>
-      <h2>- Projects List -</h2>
+      <Search
```
Comment (CarVigil) [inline src/Components/Projects/index.js:66]: Serach bar must be on the right

**Review by CarVigil — CHANGES_REQUESTED** [review]
image

**Review by CarVigil — APPROVED** [review]

**Comment by FiolAxel** [comment]
image
The title of projects is missing.

**Review by FiolAxel — APPROVED** [review]

**Review by CarVigil — DISMISSED** [review]

**Review by Gearyandres94 — APPROVED** [review]
Great Job!!!

**Review by FiolAxel — APPROVED** [review]

**Review by vadalamati — DISMISSED** [review]
Errors solved. Good job!!

File: src/Components/Projects/index.js [hunk 6]
```diff
@@ -1,70 +1,111 @@
...
 
   useEffect(async () => {
     try {
+      setLoading(true);
       const response = await fetch(`${process.env.REACT_APP_API_URL}/projects`);
       const data = await response.json();
       setProjects(data.data);
+      setData(data.data);
+      setUntouchedData(data.data);
+      setLoading(false);
```
Comment (AlejandroSantini) [inline src/Components/Projects/index.js]: Delete this SetLoading

**Review by vadalamati — APPROVED** [review]
Good job!

**Review by AlejandroSantini — COMMENTED** [review]
Nice job!

**Review by AlejandroSantini — APPROVED** [review]

**Review by Gearyandres94 — APPROVED** [review]

**Pull request merged**